### PR TITLE
fix: set "X-Request-Id" header in XMLHttpRequest only in Node.js

### PR DESCRIPTION
**Pull request by avivasyuta**
- Related to #378 

Set the `X-Request-Id` request header in `XMLHttpRequestController` only for Node.js env.

### CODE REVIEW

File: src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts
Comment (kettanaito) [inline src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts:18]: The presence of `window`, alas, doesn't guarantee this is a browser process. 

Please, could you use [`is-node-process`](https://github.com/mswjs/is-node-process) instead? It's an in-house library we built that reliably distinguishes between actual browsers, browser-like processes (like Electron or JSDOM), and Node.js processes.

```js
import { isNodeProcess } from 'is-node-process'

const IS_NODE = isNodeProcess()

if (IS_NODE) {
  // the logic that sets "X-Request-Id"
}
```

**Review by kettanaito — CHANGES_REQUESTED** [review]
Thanks for opening this one! I left a comment about determining the browser process more reliably. Let's also see how this affects the existing tests, I hope we're accounting for all the possibilities here.

File: src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts
Comment (avivasyuta) [reply src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts]: No problem. I've changed the implementation via `is-node-process`

**Comment by avivasyuta** [comment]
> Thanks for opening this one! I left a comment about determining the browser process more reliably. Let's also see how this affects the existing tests, I hope we're accounting for all the possibilities here.

At least all tests passed.

**Comment by avivasyuta** [comment]
> Thanks for opening this one! I left a comment about determining the browser process more reliably. Let's also see how this affects the existing tests, I hope we're accounting for all the possibilities here.

@kettanaito could you run workflow?

**Comment by avivasyuta** [comment]
@kettanaito Hello. Could you please approve workflow and review changes?

**Comment by avivasyuta** [comment]
@kettanaito Hello. Could you please approve workflow and review changes?

**Comment by avivasyuta** [comment]
@kettanaito @msutkowski @marcosvega91 @timdeschryver could you please review this PR?

**Comment by kettanaito** [comment]
@avivasyuta, please be respectful of the maintainers' time. All of us approach open source to the best of our abilities, and if no one has looked at this pull request it simply means no one had time. I ask of you to be patient.

**Comment by kettanaito** [comment]
I've added a check that we don't send `x-request-id` during the XHR browser tests. Passes locally, waiting for the CI.

**Review by kettanaito — APPROVED** [review]
This looks good 🚀

**Comment by kettanaito** [comment]
Got some weird behavior from Playwright locally vs remote (encourages to uninstall `playwright` locally but fails without it remotely). Oh well, adding back `playwright`.

**Comment by kettanaito** [comment]
I've resolved the issue with Playwright (see #407). Will update this branch and the tests should pass then.

**Comment by kettanaito** [comment]
Welcome to contributors, @avivasyuta 👏

**Comment by avivasyuta** [comment]
> Welcome to contributors, @avivasyuta 👏

Thank you for taking the time to view this PR.

**Comment by kettanaito** [comment]
This change will be automatically released.

**Pull request merged**
